cilium/CFP-30235-xds-adapter.md

```diff
@@ -0,0 +1,204 @@
+# CFP-30235: xDS Adapter for ClusterIP Routing
+
+**SIG:** Agent
+
+**Begin Design Discussion:** 2023-01-03
+
+**Cilium Release:** 1.16
+
+**Authors:** Rob Scott <robertjscott@google.com>
+
+## Summary
+
+Add a new xDS adapter in Cilium that could take advantage of some of the
+strengths of xDS, particularly the feedback loop via LRS and the overall
+potential for scalability improvements when adjustments to routing
+configurations don't need to round trip through the Kubernetes API Server.
+This adapter would be an alternative source of endpoints and would not replace
+the existing default behavior of reading directly from Kubernetes APIs.
+
+## Motivation
+
+There are some capabilities that the Kubernetes API Server is not well equipped
+to handle. This includes topology aware routing (or any form of routing that
+requires a feedback loop) and routing to endpoints outside of the cluster.
+
+## Goals
+
+* Describe how an xDS adapter could work in Cilium
+* Highlight new capabilities that xDS could enable
+
```

Comment:
Would a requirement/goal be the proposal could be extended for other scenarios using xDS without major changes?

I echo comments from @mikemorris on slack: `the intersection is relevant to the extent  that we just try to avoid boxing ourselves in by making the ClusterIP xDS routing too narrow/bespoke such that it would prevent further extensibility later`

Reply:
Good point - added the following goal: "Provide room for future growth in xDS support beyond ClusterIP routing."

Reply:
Thank you Rob!!